op-batcher/batcher/driver.go

```diff
@@ -547,16 +592,86 @@ func (l *BatchSubmitter) throttlingLoop(wg *sync.WaitGroup, pendingBytesUpdated
 	retryTimer := time.NewTimer(retryInterval)
 	retryTimer.Stop()
 
+	// Initialize map to store all throttling client connections
+	throttlingClients := make(map[string]*rpc.Client)
+
+	// Function to connect to all configured endpoints or fallback to default
+	connectToEndpoints := func() bool {
+		endpoints := l.Config.ThrottlingEndpoints
+
+		// Clear existing connections to ensure we have a fresh set
+		for endpoint := range throttlingClients {
+			delete(throttlingClients, endpoint)
+		}
+
+		// If no throttling endpoints are configured, use the default L2 endpoint
+		if len(endpoints) == 0 {
+			ctx, cancel := context.WithTimeout(l.shutdownCtx, l.Config.NetworkTimeout)
+			cl, err := l.EndpointProvider.EthClient(ctx)
+			cancel()
+
+			if err != nil {
+				l.Log.Warn("Failed to get default L2 endpoint", "err", err)
+				return false
+			}
+
+			// Add default endpoint to our map
+			defaultEndpoint := "default-l2-endpoint"
+			throttlingClients[defaultEndpoint] = cl.Client()
+			l.Log.Info("Using default L2 endpoint for throttling", "endpoint", defaultEndpoint)
+			return true
+		}
+
+		// Process configured throttling endpoints
+		allConnected := true
+		for _, endpoint := range endpoints {
+			client, err := rpc.Dial(endpoint)
+			if err != nil {
+				l.Log.Warn("Failed to connect to throttling endpoint", "endpoint", endpoint, "err", err)
+				allConnected = false
+				continue
+			}
+			throttlingClients[endpoint] = client
+			l.Log.Info("Connected to endpoint for throttling distribution", "endpoint", endpoint)
+		}
+
+		// Only return true if we connected to all endpoints
+		if !allConnected {
+			// Clean up any connections we did make
+			for endpoint := range throttlingClients {
+				delete(throttlingClients, endpoint)
+			}
+			return false
+		}
+
+		return len(throttlingClients) > 0
+	}
+
+	// Initial connection to endpoints
+	endpointsConnected := connectToEndpoints()
+	if !endpointsConnected {
+		l.Log.Warn("Could not connect to all throttling endpoints, will retry periodically")
+		retryTimer.Reset(retryInterval)
+	}
+
 	updateParams := func(pendingBytes int64) {
 		retryTimer.Stop()
+
+		// If we have no endpoint connections, try to connect again
+		if len(throttlingClients) == 0 {
+			l.Log.Warn("No throttling endpoints available, attempting to reconnect")
+			endpointsConnected = connectToEndpoints()
+			if !endpointsConnected {
+				l.Log.Warn("Still could not connect to all throttling endpoints, will retry")
+				retryTimer.Reset(retryInterval)
+				return
+			}
+		}
+
 		ctx, cancel := context.WithTimeout(l.shutdownCtx, l.Config.NetworkTimeout)
 		defer cancel()
-		cl, err := l.EndpointProvider.EthClient(ctx)
```

Comment:
So do I see correctly that throttling is now _only_ happening on the endpoints provided with `throttling-endpoints`? Is that intended, so that the `l2-eth-rpc` endpoints are not throttled, if not also added in this list?
This may make sense, but would it maybe also make sense to set the `ThrottlingEndpoints` to the `l2-eth-rpc` endpoints if the `throttling-endpoints` flag wasn't specified? In a normal setup without special throttling endpoints, all sequencer should probably be throttled. This would also retain the current behavior, at least more or less besides the difference that previously, we only throttled the active sequencer, now all.